eng/scripts/job-matrix/test-matrix-functions.tests.ps1

```diff
@@ -1,8 +1,7 @@
+Using Module ./test-matrix-functions.psm1
```

Comment:
Is there a reason to use a PS module here? In general we have moved away from using PS modules for our helper scripts as they tend to be more trouble then they are worth. If there isn't a clear reason I'd suggest just dot-sourcing the file instead of importing it as a module.

Reply:
Import-Module and dot sourcing don't import class definitions, which I use in the tests for type assertions and casting of test objects. I could write the code without these classes (one of which is used for config validation, and the other as a type shortcut, but there's so much various object marshalling here that it's getting unwieldy to maintain without introducing types.

Reply:
Oh I miss read that I assumed this was Import-Module. I don't think I've used the "using module" feature before, not exactly sure what it does.

Reply:
Does it have to be a module to support the class import? If not I still think it might be better to use ps1 instead of psm1.

Reply:
Yeah it seems to fail when run against a `ps1`, hence the rename:

```
$ ls
test-matrix-functions.ps1  test-matrix-functions.psm1
$ using module ./test-matrix-functions.ps1
InvalidOperation: /home/ben/sdk/azure-sdk-for-net/eng/scripts/job-matrix/test/test-matrix-functions.ps1:3
Line |
   3 |  class OrderedDictionary : System.Collections.Specialized.OrderedDicti …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Unable to find type [OrderedDictionary].

$ using module ./test-matrix-functions.psm1
$ get-module

ModuleType Version    PreRelease Name                                ExportedCommands
---------- -------    ---------- ----                                ----------------
Manifest   7.0.0.0               Microsoft.PowerShell.Management     {Add-Content, Clear-Content, Clear-Item, Clear-ItemProperty…}
Manifest   7.0.0.0               Microsoft.PowerShell.Utility        {Add-Member, Add-Type, Clear-Variable, Compare-Object…}
Script     2.0.2                 PSReadLine                          {Get-PSReadLineKeyHandler, Get-PSReadLineOption, Remove-PSReadLineKeyHandler, Set-PSReadLineKeyHandler…}
Script     0.0                   test-matrix-functions
Script     0.0                   test-matrix-functions               {ConvertToMatrixArrayFormat, CreateDisplayName, CreateMatrixEntry, GenerateFullMatrix…}
```

Reply:
I think `using` is new as of powershell 5.

Reply:
I could also split the class definitions out into their own separate module file, and keep all the functions dot-sourcable.